client/src/components/ExpandedRowDataEdit.tsx

```diff
@@ -0,0 +1,94 @@
+import React, {
+    MouseEvent,
+    useRef,
+    useState
+} from 'react';
+import {DataEditProps} from '../lib/types';
+import {fetchData} from '../services/fetch';
+import {logError} from '../services/log';
+
+const ExpandedRowDataEdit = ({data, onError, onUpdate}: DataEditProps) => {
+    // This reference object is use to collect updated data by individual form input control.
+    const collectedData = useRef({...data});
+
+    // Use this state to track and temporarily disabled all buttons when Save button is clicked.
+    const [saveClicked, setSaveClicked] = useState(false);
+
+    const handleSaveClicked = ({currentTarget}: MouseEvent<HTMLButtonElement>) => {
+        if (JSON.stringify(collectedData.current) !== JSON.stringify(data)) {
+            const value = currentTarget.getAttribute('value');
+            const {note, jiraTicket} = collectedData.current;
+
+            setSaveClicked(true);
+
+            fetchData(`/api/action/save/${value}`, {
+                method: 'POST',
+                headers: {
+                    'Content-Type': 'application/json'
+                },
+                body: JSON.stringify({
+                    note,
+                    jiraTicket,
+                })
+            })
+                .then((result) => {
+                    if (!result.data) {
+                        onError && onError('Error saving data, see console error log for details.');
```

Comment:
Same as the above note, is the `&&` operator just to check that the function is not undefined?